ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -0,0 +1,565 @@
+    df = trim_nans_off_rows(df)
+    df = df.transpose()
+    return df
+
+def list_instrument_nominal_depth(nc_file_list):
+    """ return a list of nominal_depth gatt from the nc files"""
+    instrument_nominal_depth = []
+    for f in nc_file_list:
+        netcdf_file_obj = Dataset(f, 'r')
+        instrument_nominal_depth.append(netcdf_file_obj.instrument_nominal_depth)
+        netcdf_file_obj.close()
+    return instrument_nominal_depth
+
+def generate_fv02_filename(df, nc_file_list):
+    """ return the file name only of the FV02 product """
+    netcdf_file_obj   = Dataset(nc_file_list[0], 'r')
+    site_code         = netcdf_file_obj.site_code
+    deployment_code   = netcdf_file_obj.deployment_code
+    input_netcdf_name = os.path.basename(nc_file_list[0])
+    pattern           = re.compile("^(IMOS_.*)_([A-Z].*)_([0-9]{8}T[0-9]{6}Z)_(.*)_FV01_(.*)_END")
+    match_group       = pattern.match(input_netcdf_name)
+
+    time_values = transform_datetime64_datetime(df.columns.values)
+    time_start  = min(time_values).strftime('%Y%m%dT%H%M%SZ')
+    time_end    = max(time_values).strftime('%Y%m%dT%H%M%SZ')
+
+    output_netcdf_name = '%s_T_%s_%s_FV02_%s_gridded_END-%s.nc' % (match_group.group(1), time_start,
+                                                                    site_code, deployment_code, time_end)
+
+    netcdf_file_obj.close()
+    return output_netcdf_name
+
+def transform_datetime64_datetime(datetime64_object):
+    """ The pandas date object is turned into a datetime64 type. This creates an issue when
+    doing a normal datetime operation which is fixed by running this function """
+    # http://stackoverflow.com/questions/13703720/converting-between-datetime-timestamp-and-datetime64
+    ns          = 1e-9 # nanoseconds
+    time_values = [datetime.utcfromtimestamp(x.astype(int) * ns) for x  in datetime64_object]
+    return time_values
+
+def generate_fv02_netcdf(df, nc_file_list):
+    """ generated the FV02 temperature gridded product netcdf file """
+    tmp_netcdf_dir          = tempfile.mkdtemp()
+    output_netcdf_file_path = os.path.join(tmp_netcdf_dir, generate_fv02_filename(df, nc_file_list))
+
+    input_netcdf_obj               = Dataset(nc_file_list[0], 'r')
+    output_netcdf_obj              = Dataset(output_netcdf_file_path, "w", format="NETCDF4")
+    output_netcdf_obj.date_created = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # read gatts from input, add them to output. Some gatts will be overwritten
+    input_gatts     = input_netcdf_obj.__dict__.keys()
+    gatt_to_dispose = ['author', 'toolbox_input_file', 'file_version', 'file_version_quality_control', 'quality_control_set',
+                       'CoordSysBuilder_', 'netcdf_filename', 'metadata', 'instrument_serial_number',
+                       'instrument_nominal_depth', 'compliance_checker_version', 'compliance_checker_last_updated',
+                       'geospatial_vertical_min', 'geospatial_vertical_max', 'featureType']
+
```

Comment:
variable to add to the list of attribute to delete: 
 data_centre and data_centre_email ,institution_references( referencing all to eMII)  . Add them to your attribute list with correct reference to AODN
ttime_deplotyment_start_origin ,time_deplotyment_end_origin   (value meaningless)

instrument_sample_interval should also be updated cause instruments on the mooring line can have different sampling interval (rare but possible)